### ISSUE TITLE
fetcher.go:90 Could not fetch cloud config Unexpected response status: 503

### ISSUE THREAD

**Issue by uaalto** [getlantern/lantern#3983]
```
{"extra":{"androidDevice":"f460","androidModel":"LG-F460","androidSdkVersion":"23 (6.0.1)","country":"CN","language":"","logLevel":"ERROR","osArch":"arm","osName":"android","osVersion":"Unknown 0.0","sessionUserAgents":"Apache-HttpClient/UNAVAILABLE (java 1.4): 1 requests; Crashlytics Android SDK/1.3.4.92: 1 requests; Go-http-client/1.1: 78 requests; ","timeZone":"UTC","version":"2.1.2 (20160413.044557)"},"fullMessage":"ERROR flashlight.config: fetcher.go:90 Could not fetch cloud config Unexpected response status: 503\n","hostname":"hidden","instanceid":"8kh6evqR","locationInfo":"ERROR flashlight.config","message":"fetcher.go:90 Could not fetch cloud config Unexpected response status: 503","timestamp":1460686230804}
```

**Comment by aranhoide** [comment]
@uaalto : please bump up the priority if this looks like it's happening all the time for some recent clients.

**Comment by uaalto** [comment]
Ok @aranhoide. The problem with this issue is that could be caused by many things, but since the config-server never replies with 503, the starting point might be the proxies. What do you think?

**Comment by myleshorton** [comment]
Hey @uaalto and @aranhoide I was just looking at config server logs, and it replies with 503 on request timeouts. There have been about 1000 of those in the last day:

```
desc="Request timeout" method=GET path="/cloud.yaml.gz?a4a354e6-a672-4fe7-bc85-7a6cde79f909" status=503 bytes=0 
```

**Comment by aranhoide** [comment]
That will do it!  Request timeouts are a known issue ATM.  Currently one of my immediate tasks is moving the config-server to DO.  I'll try and do some profiling in a test DO instance, both to solve these and to try and find optimization opportunities.

**Comment by myleshorton** [comment]
Reading my mind @aranhoide! Still not a lot of timeouts considering the request volume -- I was mostly just trying to diagnose why we're seeing those 503s at all.

I also really think caching would make a huge difference and would be actually quite easy to implement. I've used ehcache before, and it is incredibly easy to set up. There are other options now too like Apache Commons JCS and Google Guava Cache, but ehcache is super battle tested.

Quick question: this will just run in a single larger instance on DO right?

I was just looking at these:

http://www.ehcache.org/documentation/3.0/getting-started.html
https://github.com/alienscience/cache-dot-clj/blob/master/ehcache/src/cache_dot_clj/ehcache.clj

**Comment by uaalto** [comment]
If we are moving the config-server to DO (which I advocated for, as you might remember from my rant about Heroku limiting our instrumentation options), we have to keep in mind that this will also affect the upcomin Pro finalization Epic. For example, there is this PR: https://github.com/getlantern/config-server/pull/43 for supporting TLS for Redis that needs to be synchronized with the pro-server.
One of the nice things of having the config-server in Heroku is that it is the same ecosystem than the Pro server. But we could move also the Pro server to DO following the same logic, which might be a good decision before deployment in production.

**Comment by myleshorton** [comment]
Good point @uaalto. Yeah I think we should move the pro server to DO as well.

**Comment by myleshorton** [comment]
The current borda version of this same error is as follows. Note it's in `chainedandfronted`:

```
origin: config.getiantem.org
osArch: 386
osVersion: Windows 8 / Windows Server 2012
go_version: go1.6.2_lantern_20160502
language: zh
http_proto: HTTP/1.1
app: lantern-client
hostname: hidden
osName: windows
country: CN
http_response_status_code: 503
version: 3.1.4 (20160830.192023)
http_method: GET
root_op: fetch_config
timeZone: CST
locale_country: CN
op: chainedandfronted
fullMessage: unsuccessful response status
locationInfo: flashlight.proxied: proxied.go:155
message: unsuccessful response status
```
